pyspark-interpreter/src/main/scala/org/apache/toree/kernel/interpreter/pyspark/PySparkInterpreter.scala

```diff
@@ -155,11 +154,22 @@ class PySparkInterpreter(
   // Unsupported
   override def doQuietly[T](body: => T): T = ???
 
-  // TODO Identify how to plumb python version to here
-  override def languageInfo = LanguageInfo(
-    "python",
-    pySparkState.getVersion(),
-    fileExtension = Some(".py"),
-    pygmentsLexer = Some("ipython2"))
+  override def languageInfo: LanguageInfo = {
+    if (!pySparkService.isRunning) pySparkService.start()
 
+    import scala.util.control.Breaks._
+    val waitLimit = System.currentTimeMillis() + java.util.concurrent.TimeUnit.SECONDS.toMillis(5)
+    while (!pySparkState.isReady) {
```

Comment:
Now that I see this, it's not good to have a loop like this (even if the max timeout is  5 milliseconds). So, two comments with this:

1. Move the `java.util.concurrent.TimeUnit.SECONDS.toMillis(5)` to a constant. Maybe just create a PySparkInterpreter companion object and put the constant in there.
2. Add a `Thread.sleep(1)` or something to your while loop at the end. This a) frees up the CPU to handle other tasks and b) enables the while loop to be interruptible.

If you add those two things, I suppose we can add this for now, but it does feel like a hack.